java/code/src/com/suse/manager/webui/services/pillar/MinionPillarFileManager.java

```diff
@@ -61,24 +52,14 @@ public MinionPillarFileManager(MinionPillarGenerator minionPillarGeneratorIn) {
      */
     public void updatePillarFile(MinionServer minion) {
         this.minionPillarGenerator.generatePillarData(minion).ifPresentOrElse(
-                (pillar) -> this.saveFileToDisk(pillar, this.minionPillarGenerator.getFilename(minion.getMinionId())),
+                (pillar) -> {
+                    minion.getPillars().add(pillar);
```

Comment:
Isn't this redundant? Minion pillar is modified already in generatePillarData.

Reply:
right